runtime/lua/vim/filetype/detect.lua

```diff
@@ -0,0 +1,31 @@
+local M = {}
+
+-- This function checks the first 15 lines for appearance of 'FoamFile'
+-- and then 'object' in a following line.
+-- In that case, it's probably an OpenFOAM file
+function M.foam()
+  local foam_file = false
+  for line = 1, 15 do
+    local current_line = vim.fn.getline(line)
+    if current_line:find("^FoamFile") then
+      foam_file = true
+    elseif foam_file and current_line:find("^%s*object") then
+      vim.cmd("setf foam")
+    end
+  end
+end
+
+-- Determine if a *.tf file is TF mud client or terraform
+function M.tf()
+  local number_of_lines = vim.fn.line("$")
+  for line = 1, number_of_lines do
+    local current_line = vim.fn.trim(vim.fn.getline(line))
+    if not current_line:find("^[;/ ]") then
+      vim.cmd([[setf terraform]])
+      return
+    end
+    vim.cmd([[setf tf]])
+  end
+end
```

Comment:
All the comments I made above apply here too, but I know this is not your code :) Ping @dundargoc.